pkg/image/registry/imagestream/strategy.go

```diff
@@ -156,12 +156,6 @@ func (s Strategy) tagsChanged(old, stream *api.ImageStream) fielderrors.Validati
 		}
 
 		if tagRef.From.Kind == "DockerImage" && len(tagRef.From.Name) > 0 {
-			event, err := tagReferenceToTagEvent(stream, tagRef, "")
-			if err != nil {
-				errs = append(errs, fielderrors.NewFieldInvalid(fmt.Sprintf("spec.tags[%s].from", tag), tagRef.From, err.Error()))
-				continue
-			}
-			api.AddTagEventToImageStream(stream, tag, *event)
```

Comment:
Three use cases:

1. Import an entire registry (all tags)
2. Reference a different tag without metadata retrieval
3. Import a tag with metadata retrieval

1 and 3 are the most likely, neither 2 nor 3 can result in two stream updates.  Users need to be able to clearly define reference vs import